### DISCUSSION TITLE
Hosting on subdomain with Apache

### DISCUSSION THREAD

**Discussion by marian-code** [firefly-iii/firefly-iii#6455] [Firefly III] [answered]
Hi, I can't seem to figure out the right configuration for Apache reverse proxy to run Firefly OK.

I have Firefly installed the standard way with docker exposed on port `8080`. I can reach it fine on `http://localhost:8080/`. I set the initial configuration this way.

After this I have set apache reverse proxy to be able to access it through https. Accessing it through `https://firefly.mydomain.com`  gets me to the login screen but after I fill in the data it refuses to log in. In the dev tool console I see:

```
Refused to send form data to 'http://firefly.mydomain.com/login' because it violates the following Content Security Policy directive: "form-action 'self'"
```
And indeed looking at the sources I see **http** not **https** addresses.

```html
<form action="http://firefly.mydomain.com/login" ...>
```

My apache config looks like this:

```apache
<VirtualHost *:443>
        ServerName firefly.mydomain.com
        ServerAdmin EMAIL
        ProxyPreserveHost On
        ProxyRequests Off
        SSLEngine On
        SSLCertificateFile      /etc/letsencrypt/live/.../fullchain.pem
        SSLCertificateKeyFile   /etc/letsencrypt/live/.../privkey.pem
        ProxyPass / http://127.0.0.1:8080/
        ProxyPassReverse / http://127.0.0.1:8080/
        ErrorLog ${APACHE_LOG_DIR}/finance_error.log
        CustomLog ${APACHE_LOG_DIR}/finance_access.log combined
</VirtualHost>
```

There are simillar Issues/discussions here but all concern NGINX/Traeffik ... which I am not familiar with. This is probably the same issue as: #4991

**Comment by JC5** [comment]
I'll have to Google this, which means you can as well. You need to set these headers using Apache:

```
		 X-Forwarded-Host $host;
		 X-Forwarded-Server $host;
		 X-Forwarded-Proto $scheme;
		 X-Forwarded-For $proxy_add_x_forwarded_for;
		 Host $host;
```

**Comment by marian-code** [comment, marked as answer]
Thanks for the fast reply. Its not always easy to translate NginX config to Apache, especially when one has only limited knowledge of how they work. In the end I found the answer, so I am posting it here. Maybe somebody will find it usefull.

By default the `ProxyPass` directive of Apache passes these headers:
```apache
X-Forwarded-For # The IP address of the client.
X-Forwarded-Host # The original host requested by the client in the Host HTTP request header.
X-Forwarded-Server # The hostname of the proxy server.
```

Apart from that you have to manually set:
```apache
RequestHeader set X-Forwarded-Proto expr=%{REQUEST_SCHEME}
RequestHeader set X-Forwarded-SSL expr=%{HTTPS}
```
As discussed in this [link](https://webmasters.stackexchange.com/questions/97005/setting-x-forwarded-proto-under-apache-2-4). So finall config will look like this:
```apache
<VirtualHost *:443>
        ServerName firefly.mydomain.com
        ServerAdmin EMAIL
        ProxyPreserveHost On
        ProxyRequests Off
        SSLEngine On
        SSLCertificateFile      /etc/letsencrypt/live/.../fullchain.pem
        SSLCertificateKeyFile   /etc/letsencrypt/live/.../privkey.pem
        ProxyPass / http://127.0.0.1:8080/
        ProxyPassReverse / http://127.0.0.1:8080/
        ErrorLog ${APACHE_LOG_DIR}/finance_error.log
        CustomLog ${APACHE_LOG_DIR}/finance_access.log combined
        RequestHeader set X-Forwarded-Proto expr=%{REQUEST_SCHEME}
        RequestHeader set X-Forwarded-SSL expr=%{HTTPS}
</VirtualHost>
```

**Reply by JC5** [reply]
Thanks for sharing. I'll put this in the documentation as well.
